packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -829,22 +829,27 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     }
 
     // Align items along the main axis.
-    final double idealSize = canFlex && mainAxisSize == MainAxisSize.max ? maxMainSize : allocatedSize;
+    // Unfortunately, using full precision floating point here causes false
+    // positives on the overflow detection.
+    // We apply a similar hack here as _applyFloatingPointHack in
+    // text_painter.dart.
+    final double idealSize = canFlex && mainAxisSize == MainAxisSize.max
+      ? maxMainSize.ceilToDouble()
```

Comment:
Would it make sense to factor this out into a _applyFloatingPointHack method like in test_painter.dart? That why it's immediately clear in each line that this is a hack

Reply:
Ended up changing this to just check an epsilon rather than truncate.